Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -539,37 +539,16 @@ namespace OpenInfraPlatform {
 								if(hollow->InnerFilletRadius) {
 									r2 = hollow->InnerFilletRadius * length_factor;
 								}
-
 								// Outer
-								if(r1 != 0) {
-									addArc(outer_loop, r1, 0, M_PI_2, x * 0.5 - r1, y * 0.5 - r1);
-									addArc(outer_loop, r1, M_PI_2, M_PI_2, -x * 0.5 + r1, y * 0.5 - r1);
-									addArc(outer_loop, r1, M_PI, M_PI_2, -x * 0.5 + r1, -y * 0.5 + r1);
-									addArc(outer_loop, r1, 3 * M_PI_2, M_PI_2, x * 0.5 - r1, -y * 0.5 + r1);
-								}
-								else {
-									outer_loop.push_back(carve::geom::VECTOR(-x * 0.5, -y * 0.5));
-									outer_loop.push_back(carve::geom::VECTOR(x * 0.5, -y * 0.5));
-									outer_loop.push_back(carve::geom::VECTOR(x * 0.5, y * 0.5));
-									outer_loop.push_back(carve::geom::VECTOR(-x * 0.5, y * 0.5));
-								}
-
+								GetRectangleCoordinates(outer_loop, r1, x, y);
+								
+							
 								// Inner
 								std::vector<carve::geom::vector<2>> inner_loop;
 								x -= 2 * t;

```

Comment:
In order to not confuse compiler with implicit casting between int and double, use doubles in calculations with doubles.
```suggestion
								x -= 2. * t;
```

Reply:
Done